Add method: CellMapper

Marius1311 opened the pull request:
## Describe your changes

Add new method for label projection: CellMapper. See https://cellmapper.readthedocs.io/en/latest/

## Checklist before requesting a review
- [x] I have performed a self-review of my code

- Check the correct box. Does this PR contain:
  - [ ] Breaking changes
  - [x] New functionality
  - [ ] Major changes
  - [ ] Minor changes
  - [ ] Bug fixes

- [ ] Proposed changes are described in the CHANGELOG.md

- [ ] CI Tests succeed and look good!

lazappi reviewed (commented):
Looks pretty good to me apart from some minor things. Can you confirm if someone has tried running the test workflow?

@Marius1311 you could also add yourself as a contributor in the `_viash.yaml` file if you like.

Marius1311 commented on CHANGELOG.md:
```diff
@@ -1,5 +1,6 @@
 # task_label_projection devel
 
+* Add new method: CellMapper (PR #22)
```
sure, will do!

Marius1311 commented on src/methods/cellmapper_scvi/config.vsh.yaml:
```diff
@@ -0,0 +1,67 @@
+      kernel_method: gauss
+    
+arguments:
+  - name: "--kernel_method"
+    type: "string"
+    choices: ["hnoca", "gauss"]
+    default: "hnoca"
+    description: Kernel function to compute k-NN edge weights. 
+  - name: "--n_neighbors"
+    type: "integer"
+    default: 30
+    description: Number of neighbors to consider for k-NN graph construction.
+  - name: "--num_hvg"
+    type: integer
+    description: "The number of HVG genes to subset to."
+  - name: --n_latent
+    type: integer
+    default: 30
+    description: Number of latent dimensions.
+  - name: --n_hidden
+    type: integer
+    default: 128
+    description: Number of hidden units.
+  - name: --n_layers
+    type: integer
+    default: 2
+    description: Number of layers.
```
thanks, I added this to each parameter's description.

Marius1311 replied:
ok, done!

Marius1311 commented:
Hi @lazappi, thanks for the review, I extended the description in the Changelog, updated parameter descriptions, and added myself as a contributor. Let me know whether there's anything else I can do!

rcannood reviewed (commented):
LGTM as far as I'm concerned! @lazappi Is this also ok for you?
